buying_selling/posts/api/views.py

```diff
@@ -108,6 +108,17 @@ def retrieve(self, request, pk, format=None):
         post_serializer['images'] = images
         return Response(post_serializer)
 
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True, context={'request': request})
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
```

Comment:
Why was this block required? Earlier it used to work without it.

Reply:
Yes, it is required since we have to pass the request back to the serializer, in order to build the absolute URL for the image, else the image URL would look something like media/wallpaper.png

Reply:
Okay